### PR TITLE
Brought warning to higher priority than note, etc.

### PR DESCRIPTION
**Pull request by Meshugah**
#5857 

Update functions.rst

Update for warning over note priority

Updated units-and-global-variables.rst

Updated file for warning over note priority

Updated control-structures.rst

Updated priority of warning over note

Updated assembly.rst

Updated priority of warning over code

Updated introduction-to-smart-contracts.rst

Updated priority of warning over note

Update installing-solidity.rst

Corrected control-structures.rst

Corrected white space error

Corrected whitespace

Corrected whitespace introduction to smart contract

### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [x] New tests have been created which fail without the change (if possible)
- [x] README / documentation was extended, if necessary
- [x] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

### CODE REVIEW

**Comment by Meshugah** [comment]
@ChrisChinchilla Let me know if I can improve on this further somehow 😄

File: docs/assembly.rst [hunk 1]
```diff
@@ -394,6 +394,16 @@ use ``x_slot``, and to retrieve the byte-offset you use ``x_offset``.
...
 
+.. warning::
+    If you access variables of a type that spans less than 256 bits
+    (for example ``uint64``, ``address``, ``bytes16`` or ``byte``),
+    you cannot make any assumptions about bits not part of the
+    encoding of the type. Especially, do not assume them to be zero.
+    To be safe, always clear the data properly before you use it
+    in a context where this is important:
+    ``uint32 x = f(); assembly { x := and(x, 0xffffffff) /* now use x */ }``
+    To clean signed types, you can use the ``signextend`` opcode.
```
Comment (bshastry) [inline docs/assembly.rst]: Would it be possible to provide an example for the use of `signextend` to clear higher order bits?

Comment (Meshugah) [reply docs/assembly.rst]: I've added an example. Is this correct and would it be what you've requested?

File: docs/assembly.rst [hunk 2]
```diff
@@ -394,6 +394,17 @@ use ``x_slot``, and to retrieve the byte-offset you use ``x_offset``.
...
+.. warning::
+    If you access variables of a type that spans less than 256 bits
+    (for example ``uint64``, ``address``, ``bytes16`` or ``byte``),
+    you cannot make any assumptions about bits not part of the
+    encoding of the type. Especially, do not assume them to be zero.
+    To be safe, always clear the data properly before you use it
+    in a context where this is important:
+    ``uint32 x = f(); assembly { x := and(x, 0xffffffff) /* now use x */ }``
+    To clean signed types, you can use the ``signextend`` opcode:
+    ``assembly { signextend(0, x) }``
```
Comment (bshastry) [inline docs/assembly.rst]: I'm not sure if this is right: 

```
signextend(i, x): sign extend from (i*8+7)th bit counting from least significant
```

so signextend(0,x) will sign extend from the 7th bit from LSB. What happens if you have a signed 32 bit integer (int32). Wouldn't it change the value itself?

Perhaps, this is what you meant?

```
assembly { signextend(<bit_width_of_x>, x) }
```

**Comment by ChrisChinchilla** [comment]
@bshastry This was primarily a formatting PR, if there were errors in the original text, changes might be better in another PR, but I'll check through anyway.

File: docs/assembly.rst (hunk 2 above)
Comment (ChrisChinchilla) [reply docs/assembly.rst]: @Meshugah Purely for readability, I think this warning block is better under the code block.

**Review by ChrisChinchilla — CHANGES_REQUESTED** [review]
From my perspective, the one change I requested aside, this is just what I was looking for @Meshugah

**Review by ChrisChinchilla — APPROVED** [review]
Looks good to me! Thanks for your work @Meshugah

File: docs/control-structures.rst [hunk 3]
```diff
@@ -66,6 +66,28 @@ as the actual contract has not been created yet.
 Functions of other contracts have to be called externally. For an external call,
 all function arguments have to be copied to memory.
 
+.. warning::
+  Be careful that ``feed.info.value(10).gas(800)`` only locally sets the ``value`` and amount of ``gas`` sent with the function call, and the parentheses at the end perform the actual call. So in this case, the function is not called and the ``value`` and ``gas`` settings are lost.
```
Comment (chriseth) [inline docs/control-structures.rst:70]: Please undo the move. `.value` is only introduced later on.

File: docs/control-structures.rst [hunk 4]
```diff
@@ -385,6 +385,9 @@ There are two other ways to trigger exceptions: The ``revert`` function can be u
 revert the current call. It is possible to provide a string message containing details about the error
 that will be passed back to the caller.
 
+.. warning::
```
Comment (chriseth) [inline docs/control-structures.rst:388]: Please undo this move, it refers to the paragraph after the note.

File: docs/installing-solidity.rst [hunk 5]
```diff
@@ -298,16 +298,16 @@ You might want to install ccache to speed up repeated builds.
 CMake will pick it up automatically.
 Building Solidity is quite similar on Linux, macOS and other Unices:
 
+.. warning::
```
Comment (chriseth) [inline docs/installing-solidity.rst:301]: This should be moved further down, it distracts the reading flow.

File: docs/units-and-global-variables.rst [hunk 6]
```diff
@@ -163,6 +163,16 @@ Mathematical and Cryptographic Functions
 ``keccak256(bytes memory) returns (bytes32)``:
     compute the Keccak-256 hash of the input
 
+.. warning::
```
Comment (chriseth) [inline docs/units-and-global-variables.rst:166]: Please undo this move, it refers to ecrecover, which is only introduced further down.

File: docs/control-structures.rst (hunk 3 above)
Comment (ChrisChinchilla) [reply docs/control-structures.rst:70]: Looks like this was fixed by https://github.com/ethereum/solidity/pull/6846 as in `develop` it's in the right place.

File: docs/control-structures.rst (hunk 4 above)
Comment (ChrisChinchilla) [reply docs/control-structures.rst:388]: Looks like this was fixed by #6846 as in develop it's in the right place now.

**Pull request merged**